osu.Game/OsuGame.cs

```diff
@@ -307,8 +307,11 @@ private void beatmapChanged(ValueChangedEvent<WorkingBeatmap> beatmap)
             if (nextBeatmap?.Track != null)
                 nextBeatmap.Track.Completed += currentTrackCompleted;
 
-            beatmap.OldValue?.Dispose();
+            var oldBeatmap = beatmap.OldValue;
```

Comment:
`using (var oldBeatmap = beatmap.OldValue)`
This will allow you to remove `Dispose` call below.